### PR TITLE
Update toolbar to use OnPush change detection

### PR DESCRIPTION
**Pull request by luisb90**
Updated the toolbar to add OnPush change detection and added unit tests

### CODE REVIEW

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts [hunk 1]
```diff
@@ -83,7 +48,8 @@ export class ToolbarComponent {
   }
   constructor(private renderer: Renderer2, private elRef: ElementRef) {}
 
-  onMenuClicked(item, $event) {
+  onMenuClicked(item: ToolbarMenuItem, $event: any) {
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: can we change this event and its emitter to be type `Event`?

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/fixtures/toolbar.fixture.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Component, ViewChild, ChangeDetectionStrategy } from '@angular/core';
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/toolbar/fixtures/toolbar.fixture.ts]: I like that you broke this into its own file, will update my code to do the same.

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts [hunk 3]
```diff
@@ -83,7 +48,8 @@ export class ToolbarComponent {
...
+    /* istanbul ignore else */
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: we should either test this path and remove the ignore, or remove the if statement if `item.click` is not nullable

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts [hunk 4]
```diff
@@ -83,7 +48,8 @@ export class ToolbarComponent {
   }
```
Comment (aacebo) [inline projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: I think you forgot to put a new line between the end curly and the constructor.

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts (hunk 1 above)
Comment (luisb90) [reply projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: typed it up

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts (hunk 3 above)
Comment (luisb90) [reply projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: I added a test thatll test if the click function was not defined for a menu item

File: projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts (hunk 4 above)
Comment (luisb90) [reply projects/swimlane/ngx-ui/src/lib/components/toolbar/toolbar.component.ts]: woops, updated

**Pull request merged**